airbyte-container-orchestrator/src/main/java/io/airbyte/container_orchestrator/JobOrchestrator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.container_orchestrator;
+
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.scheduler.models.JobRunConfig;
+import io.airbyte.workers.temporal.sync.OrchestratorConstants;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public interface JobOrchestrator<INPUT> {
+
+  String getOrchestratorName();
+
+  Class<INPUT> getInputClass();
+
+  default INPUT readInput() throws IOException {
```

Comment:
the rest of the methods should also have a docstring since the methods in this interface aren't straightforward.